### ISSUE TITLE
Unified Driver for platform.

### ISSUE THREAD

**Issue by willwade** [nateshmbhat/pyttsx3#336]
Here's a suggestion. There is a need to have a AVSpeechSynth driver for MacOS as Nss is depreciated but also it would allow this tts wrapper to support ssml across the board.  

And on windows we also need uwp abd sapi4 driver. To make things easier I'm suggesting the default would be a unified driver for each platform
 

This is thrown together code but you can see what I'm thinking https://gist.github.com/willwade/86ef5c0655badc90c2f3fec811c06cde

**Comment by willwade** [comment]
#121 #287 #104

**Comment by cclauss** [comment]
SSML: https://cloud.google.com/text-to-speech/docs/ssml

**Comment by willwade** [comment]
> SSML: https://cloud.google.com/text-to-speech/docs/ssml

Sapi. AVSynyh & espeak all support ssml. It's on my radar to do a pr for supporting this as soon as espeak and avsynth done

**Comment by willwade** [comment]
So I’m actually wondering if windows should have a second driver using the more modern .net api. 

Here’s a quick thrown together attempt https://gist.github.com/willwade/cc268ef0b0b0ec106e644eec6835b228

it Would remove problems like #263

**Comment by cclauss** [comment]
That sound very promising!

Please name the file `system_speech.py` instead of `SystemSpeech.py` to align with Python naming.

**Comment by willwade** [comment]
> That sound very promising!
> 
> Please name the file `system_speech.py` instead of `SystemSpeech.py` to align with Python naming.
> 
> * https://docs.astral.sh/ruff/rules/#pep8-naming-n

Yeah. I’m not sure I like the name of that engine. It’s technically the .NET interface. Dotnetspeech synth” I don’t know 🤷

**Comment by cclauss** [comment]
`DotNetSpeech`, `DotNetSpeechEngine`, `DotNetSpeechDriver`
